### PR TITLE
Update register-root-component.md

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why 

Improve the definition and drop outdated info. We reference these updated in #18613

### CODE REVIEW

**Review by Simek — APPROVED** [review]
That's great, I had updating this page and removing old references on the list, thanks! 🙏

File: docs/pages/versions/unversioned/sdk/register-root-component.md [hunk 1]
```diff
@@ -7,27 +7,60 @@ packageName: 'expo'
 import {APIInstallSection} from '~/components/plugins/InstallSection';
 import PlatformsSection from '~/components/plugins/PlatformsSection';
 
-This function tells Expo what component to use as the root component for your app.
+Sets the initial React component to render natively in the app's root React Native view on iOS, Android, and web.
+Also adds dev-only debugging tools for use with `npx expo start`.
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/register-root-component.md:11]:
```suggestion
Sets the initial React component to render natively in the app's root React Native view on iOS, Android, and the web. It also adds dev-only debugging tools for use with `npx expo start`.
```

File: docs/pages/versions/unversioned/sdk/register-root-component.md [hunk 2]
```diff
@@ -37,30 +70,22 @@ Sets the main component for Expo to use for your app.
...
+
 You can set the `"main"` in **package.json** to any file within your
 project. If you do this, then you need to use `registerRootComponent`;
-`export default` will not make this component the root for the Expo app
+`export default` will not make this component the root for the app
 if you are using a custom entry file.
 
 For example, let's say you want to make `"src/main.js"` the entry file
 for your app -- maybe you don't like having JavaScript files in the
 project root, for example. First, set this in **package.json**:
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/register-root-component.md:86]:
```suggestion
For example, let's say you want to make `"src/main.js"` the entry file
for your app -- maybe you don't like having JavaScript files in the
project root. First, set this in **package.json**:
```

Not related to this PR, but the use "for example" phrase is duplicated here. I think we can remove the second one.

**Review by amandeepmittal — APPROVED** [review]
Left a few suggestions but rest looks good to me 👍 

Also, can you please apply these changes to the latest SDK 😄?

File: docs/pages/versions/unversioned/sdk/register-root-component.md (hunk 2 above)
Comment (EvanBacon) [reply docs/pages/versions/unversioned/sdk/register-root-component.md:86]: 😅

**Pull request merged**